### PR TITLE
Add tests

### LINKED ISSUE #1: Add tests
Initial implementation had some bare bones tests, so I ripped them out in order to not procrastinate publishing this any longer. This issue is to rewrite those tests. Assistance welcomed!

### PR DESCRIPTION
**Pull request by wKovacs64**
Closes #1. I'm sure it's not perfect, but it's a start.

```
=============================== Coverage summary ===============================
Statements   : 100% ( 34/34 )
Branches     : 96.43% ( 27/28 )
Functions    : 100% ( 9/9 )
Lines        : 100% ( 34/34 )
================================================================================
```

Thoughts, @tizmagik ?

### CODE REVIEW

**Comment by tizmagik** [comment]
@wKovacs64  Very nice, thank you for this. I'll try and do a full review later today or tomorrow. Could you just add a test for the `@decorator` syntax style please? 

A couple other things to really polish this off (feel free to submit separate issues to do these in follow-on PRs, that's totally fair!)
- Travis build step for PRs
- Coveralls to ensure code coverage doesn't drop
- Pre-publish test to ensure tests pass before npm publish

**Comment by tizmagik** [comment]
Please also add yourself as a contributor to the package.json -- you'll get commit access to this repo after this PR is merged. My rule is 2 contributions grants commit access.

**Comment by wKovacs64** [comment]
Sure, I can add that stuff. I didn't test decorator syntax at first because it felt like we would just be testing Babel's transform, but I'll add one.

File: test/index.test.js
Comment (tizmagik) [inline test/index.test.js]: `to.be.false`?

Why do you expect it to not be called though? Isn't it being unmounted?

File: test/setup.js
Comment (tizmagik) [inline test/setup.js]: Maybe this file isn't exactly a "setup" but more of a "TestUtil" or similar might be a better name?

File: .travis.yml
Comment (tizmagik) [inline .travis.yml:7]: 6?

File: src/insert-style.js
Comment (tizmagik) [inline src/insert-style.js:1]: Can you help me understand why this file needed to change?

**Review by tizmagik — CHANGES_REQUESTED** [review]

File: src/insert-style.js
Comment (tizmagik) [reply src/insert-style.js:1]: To clarify, I actually like the way you've refactored it, but I just want to understand if this was a necessary change to facilitate the tests or what exactly?

File: test/index.test.js
Comment (wKovacs64) [reply test/index.test.js]: Good catch, thanks.

It's being unmounted once, but it's been mounted twice (once in the initial `renderIntoDocument` call, and again in the `rendered.componentWillMount` call (manually triggered lifecycle method). So we expect the cache to remain in tact (until the last instance of it has been unmounted). We would have to unmount one more time in order for the cache to clear.

File: test/setup.js
Comment (wKovacs64) [reply test/setup.js]: Sure. I never know what to call it. :)

File: .travis.yml
Comment (wKovacs64) [reply .travis.yml:7]: "node" = current, which is 6, but we can future proof it a bit by adding 6 explicitly. I'll add it.

File: src/insert-style.js
Comment (wKovacs64) [reply src/insert-style.js:1]: Are you referring to flipping the IE8 detection conditionals? I put a note about it in the commit message but I've been rewriting history here like crazy for the PR so it's easy to miss. :) 

The 'textContent' property of a jsdom-created style element always seems to be
present (even after attempting to delete), so testing the IE8 case was proving
difficult. Inverting the conditional makes it much easier, as we're now able to
pass in a phony IE8 style element by pre-populating styleSheet.cssText.

File: test/index.test.js
Comment (tizmagik) [reply test/index.test.js]: Ah ok got it! That wasn't immediately obvious to me so maybe a comment here would help.

File: src/insert-style.js
Comment (tizmagik) [reply src/insert-style.js:1]: 👍

**Comment by wKovacs64** [comment]
Rebased and added/updated the things you mentioned. We might need to mess with some dependencies and add peers (Node 4 is mad), but I'll wait and look into it after this gets merged. I had to add `babel-runtime` to use the decorator transform, but I think it probably should have been included in #11 anyway (if I'm reading [the docs](https://babeljs.io/docs/plugins/transform-runtime) correctly).

Let me know if you want any of these squashed.

File: test/index.test.js
Comment (wKovacs64) [reply test/index.test.js]: Added a comment.

**Comment by tizmagik** [comment]
This is great, thanks @wKovacs64 !

I wouldn't worry too much about node 4, in fact, you can probably remove it if it doesn't work after merge.

**Review by tizmagik — APPROVED** [review]

**Pull request merged**
